### PR TITLE
Mention how Control focus loss affects `Viewport.gui_focus_change`

### LINKED ISSUE #88770: Signal "gui_focus_changed" doesn't emit null when focus is lost
### Tested versions

Reproducible in v4.2.1.stable

### System information

Godot v4.2.1.stable unknown - Arch Linux ZEN SMP PREEMPT_DYNAMIC Fri, 23 Feb 2024 16:32:48 +0000 - X11 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1070 (nvidia; 545.29.06) - Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz (12 Threads)

### Issue description

The results of polling `gui_get_focus_owner()` vs. tracking state via signals are different.

image

I looked into the related PR #34269 and issue #30042. It seems that at least one of these applies:

a) It's a bug.
b) The naming for the signal isn't appropriate.
c) The documentation of this behavior is missing.

From a user perspective, a signal called `gui_focus_changed` not emitting `null` on focus loss is definitely gonna be seen as a bug.

### Steps to reproduce

1. Connect the `gui_focus_changed` signal to a callback
2. Notice the callback isn't called when focus is lost

### Minimal reproduction project (MRP)

[release_focus.zip](https://github.com/godotengine/godot/files/14393764/release_focus.zip)

### PR DESCRIPTION
**Pull request by Sauermann**
resolve #88770

### CODE REVIEW

File: doc/classes/Viewport.xml [hunk 1]
```diff
@@ -414,6 +414,7 @@
 			<param index="0" name="node" type="Control" />
 			<description>
 				Emitted when a Control node grabs keyboard focus.
+				[b]Note:[/b] A Control nodes focus loss doesn't cause this signal to be emitted.
```
Comment (AThousandShips) [inline doc/classes/Viewport.xml:417]:
```suggestion
				[b]Note:[/b] A Control node's focus loss doesn't cause this signal to be emitted.
```
Or
```suggestion
				[b]Note:[/b] A Control node losing focus doesn't cause this signal to be emitted.
```

Comment (Sauermann) [reply doc/classes/Viewport.xml]: Thanks!

**Review by AThousandShips — APPROVED** [review]

**Comment by Mickeon** [comment]
Very, very minor nitpick: the commit could use a more specific name

**Review by Mickeon — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**